domain/merchant/asda.go

```diff
@@ -90,33 +98,56 @@ type asdaAPIResponse struct {
 }
 
 type asdaAPIRequest struct {
-	Data struct {
```

Comment:
@ladydascalie Given that this struct is most probably single-use (i.e. unlikely to share the same nested components as any other merchant API requests), would it not be preferable to keep as a single struct with embedded structs rather than pollute the namespace with multiple fragmented single-use structs, e.g. `ServiceInfo`, `ServiceAddress` etc.?

What happens if future merchants comprise a `"service_info"` field in their request payload which requires a different signature? (Do we end up with `AsdaServiceInfo`, `MorrisonsServiceInfo` etc.?)

Reply:
Each merchant really ought to have its own package in my opinion, this would allow for those cases and allow us to cleanly describe structs